grails-app/assets/javascripts/streama/controllers/dash-ctrl.js

```diff
@@ -13,8 +13,17 @@ angular.module('streama').controller('dashCtrl',
 
     $scope.$on('changedGenre', onChangedGenre);
 
-
-    init();
+    if(!localStorageService.get('currentProfile')){
+      apiService.profile.getUserProfiles().success(function(data) {
+          localStorageService.set('currentProfile', data[0]);
+          init();
+        }
+      ).error(function (data) {
+        alertify.error(data.message);
+      });
+    } else {
+      init();
+    }
```

Comment:
This functionality could be moved into the init() function itself (lines 16-24), that way you can just unconditionally fire the init-function right away.